pycontrails/datalib/ecmwf/era5_model_level.py

```diff
@@ -413,75 +415,55 @@ def _download_convert_cache_handler(
         ----------
         times : list[datetime]
             Times to download in a single MARS request.
-
-        Notes
-        -----
-        This function depends on `metview <https://metview.readthedocs.io/en/latest/python.html>`_
-        python bindings and binaries.
-
-        The lifetime of the metview import must last until processed datasets are cached
-        to avoid premature deletion of metview temporary files.
         """
-        try:
-            import metview as mv
-        except ModuleNotFoundError as exc:
-            dependencies.raise_module_not_found_error(
-                "model_level.grib_to_dataset function",
-                package_name="metview",
-                module_not_found_error=exc,
-                extra="See https://metview.readthedocs.io/en/latest/install.html for instructions.",
-            )
-        except ImportError as exc:
-            msg = "Failed to import metview"
-            raise ImportError(msg) from exc
-
         if self.cachestore is None:
             msg = "Cachestore is required to download and cache data"
             raise ValueError(msg)
 
         stack = contextlib.ExitStack()
-        request = self.mars_request(times)
+        ml_request = self.mars_request(times)
+        lnsp_request = self._mars_request_lnsp(times)
 
-        if not self.cache_grib:
-            target = stack.enter_context(temp.temp_file())
+        if not self.cache_raw:
+            ml_target = stack.enter_context(temp.temp_file())
+            lnsp_target = stack.enter_context(temp.temp_file())
         else:
-            request_str = ";".join(f"{p}:{request[p]}" for p in sorted(request.keys()))
-            name = hashlib.md5(request_str.encode()).hexdigest()
-            target = self.cachestore.path(f"era5ml-{name}.grib")
+            ml_target = _target_path(ml_request, self.cachestore)
+            lnsp_target = _target_path(lnsp_request, self.cachestore)
 
         with stack:
-            if not self.cache_grib or not self.cachestore.exists(target):
-                if not hasattr(self, "cds"):
-                    self._set_cds()
-                self.cds.retrieve("reanalysis-era5-complete", request, target)
-
-            # Read contents of GRIB file as metview Fieldset
-            LOG.debug("Opening GRIB file")
-            fs_ml = mv.read(target)
-
-            # reduce memory overhead by cacheing one timestep at a time
+            for request, target in ((ml_request, ml_target), (lnsp_request, lnsp_target)):
+                if not self.cache_raw or not self.cachestore.exists(target):
+                    if not hasattr(self, "cds"):
+                        self._set_cds()
+                    self.cds.retrieve("reanalysis-era5-complete", request, target)
```

Comment:
I just quickly added support for this. Not strictly async (because the CDS client isn't async as written), but this allows us to submit both requests simultaneously to avoid waiting in the queue for twice as long.

fcd2cd496f6bd1cb1be6bafcf4629f06b766c699